### PR TITLE
Migrate Blockwise to use taskspec

### PR DESCRIPTION
**Pull request by fjetter**
This is migrating the blockwise expressions and connected ones to the new task spec.

I ran into interesting issues while ripping out the recursion in https://github.com/dask/dask-expr/pull/1158 and wanted to do this step here first.

### CODE REVIEW

File: dask_expr/_expr.py [hunk 1]
```diff
@@ -3752,34 +3765,20 @@ def _broadcast_dep(self, dep: Expr):
...
+            if self._broadcast_dep(_expr):
+                subname = (_expr._name, 0)
             else:
-                graph[(_expr._name, index)] = _expr._task(index)
+                subname = (_expr._name, index)
+            subgraphs[subname] = _expr._task(subname, subname[1])
 
         for i, dep in enumerate(self.dependencies()):
-            graph[self._blockwise_arg(dep, index)] = "_" + str(i)
+            subgraphs[self._blockwise_arg(dep, index)] = "_" + str(i)
```
Comment (fjetter) [inline dask_expr/_expr.py:3778]: I have to admit that I don't understand what's going on here but I just went ahead and translated the existing code

**Review by phofl — APPROVED** [review]

**Review by phofl — CHANGES_REQUESTED** [review]
the failures look related

**Comment by fjetter** [comment]
hmm, there are problems when running the dask/dask tests with `to_bag` n stuff. That's frustrating

**Comment by fjetter** [comment]
Fix is https://github.com/dask/dask/pull/11495 

I will not change the CI config to test this. I can retrigger once dask/dask is merged

**Comment by fjetter** [comment]
Final problem is https://github.com/dask/dask/pull/11496

the test asserts on the pickled graph size which is blowing up with this since deduplication is not working well. That function caching layer was introduced a little prematurely so let's rip it out and deal with this later

**Pull request merged**